flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -303,14 +303,24 @@ private EnvironmentInstance() {
 			// register catalogs
 			catalogs.forEach(tableEnv::registerCatalog);
 
-			Optional<String> potentialCurrentCatalog = mergedEnv.getExecution().getCurrentCatalog();
-			if (potentialCurrentCatalog.isPresent()) {
-				tableEnv.useCatalog(potentialCurrentCatalog.get());
+			// set current catalog
+			Optional<String> potentialCatalog = Optional.empty();
```

Comment:
It might be better if we code similarly for both default catalog and default database.